src/FinalFrame.js

```diff
@@ -0,0 +1,22 @@
+class FinalFrame extends Frame {
+  constructor() {
+    super();
+    this.rollThree = 0;
+  }
+
+  roll(value) {
+    if (this.currentroll === 1) {
+      this.rollOne = value;
+      this.currentroll++;
+    } else if (this.currentroll === 2) {
+      this.rollTwo = value;
+      this.currentroll++;
+      if (!this.isStrike() || !this.isSpare()) {
+        this.endFrame = true;
```

Comment:
Why do you need this `endFrame` variable? Surely that can be a method that calculates based on the info?
Then you could have:
```js
roll(value) {
  switch(this.currentroll) {
    case 1:
      this.rollOne = value;
      break;
    case 2:
      this.rollTwo = value;
      break;
    case 3:
      this.rollThree = value;
  }
  this.currentroll++;
}

endFrame() {
  return (
      (this.currentroll === 4 && (this.isStrike() || this.isSpare()))
      || (this.currentroll === 3)
  )
}
```

Reply:
You could also have a `needsThirdRoll` method that returns `isSpare() || isStrike()` to simplify further.